x-pack/plugins/security_solution/public/common/hooks/eql/helpers.ts

```diff
@@ -10,55 +10,105 @@ import { inputsModel } from '../../../common/store';
 import { EqlSearchStrategyResponse } from '../../../../../data_enhanced/common';
 import { InspectResponse } from '../../../types';
 import { EqlPreviewResponse, Source } from './types';
-import { EqlSearchResponse } from '../../../../common/detection_engine/types';
+import { BaseHit, EqlSearchResponse } from '../../../../common/detection_engine/types';
 
 type EqlAggBuckets = Record<string, { timestamp: string; total: number }>;
 
 export const EQL_QUERY_EVENT_SIZE = 100;
 
-// Calculates which 2 min bucket segment, event should be
-// sorted into
+/**
+ * Calculates which 2 min bucket segment, event should be sorted into
+ * @param eventTimestamp The event to be bucketed timestamp
+ * @param relativeNow The timestamp we are using to calculate how far from 'now' event occurred
+ */
 export const calculateBucketForHour = (eventTimestamp: number, relativeNow: number): number => {
   const diff: number = relativeNow - eventTimestamp;
```

Comment:
```suggestion
  const diff = Math.abs(relativeNow - eventTimestamp);
  const minutes = Math.floor(diff / 60000);
```

You don't need to redefine the number type here, and if you add Math.abs, I think ordering won't matter and tests like this would pass:

```ts
    test('returns 2 if event occurred within 2 minutes of "now" but arguments are flipped', () => {
      const diff = calculateBucketForHour(
        Number(dateMath.parse('now')?.format('x')),
        Number(dateMath.parse('now-1m')?.format('x'))
      );

      expect(diff).toEqual(2);
    });

```

Reply:
Ah, nice, great catch! Adding that test and doing the same for `calculateBucketForDay`.